src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CryptoConfig.cs

```diff
@@ -330,16 +325,10 @@ public static object CreateFromName(string name, params object[] args)
             if (name == null)
                 throw new ArgumentNullException(nameof(name));
 
-            Type retvalType = null;
-            
+            Type retvalType;
+
             // Check to see if we have an application defined mapping
-            lock (s_InternalSyncObject)
```

Comment:
This lock is really bad and causes contention issues when our web app is running on E20 Azure VMs.
Perfview:
Name
--
clr!??AwareLock::Contention
+ clr!JITutil_MonContention
\|+ mscorlib.ni!CryptoConfig.CreateFromName
\|\|+ mscorlib.ni!Utils.ObjToHashAlgorithm
\|\|\|+ mscorlib.ni!RSACryptoServiceProvider.VerifyData
\|\|\| + microsoft.identitymodel.tokens!AsymmetricSignatureProvider.Verify
\|\|\|  +   system.identitymodel.tokens.jwt!JwtSecurityTokenHandler.ValidateSignature
\|\|\|  \|+   system.identitymodel.tokens.jwt!JwtSecurityTokenHandler.ValidateSignature
\|\|\|  \| + system.identitymodel.tokens.jwt!JwtSecurityTokenHandler.ValidateToken
\|\|\|  \|  +   microsoft.teamfoundation.framework.server!OAuth2AuthenticationService.ValidateToken
\|\|\|  \|   +   microsoft.teamfoundation.framework.server!OAuth2AuthenticationModule.OnAuthenticateRequest

Reply:
@sebastienros do we have any benchmark in the ASP.NET Perf lab that uses `CryptoConfig.CreateFromName` internally?